tests/cdblib_test.py

```diff
@@ -288,50 +289,66 @@ class WriterNativeInterfaceTestBase(object):
 
     def setUp(self):
         self.sio = sio = io.BytesIO()
-        self.writer = self.writer_cls(sio, hashfn=self.HASHFN)
+        self.writer = self.writer_cls(sio, hashfn=self.HASHFN, strict=True)
 
     def get_reader(self):
         self.writer.finalize()
-        return self.reader_cls(self.sio.getvalue(), hashfn=self.HASHFN)
-
-    def make_bad(self, method):
-        return partial(self.assertRaises, Exception, method)
+        return self.reader_cls(
+            self.sio.getvalue(), hashfn=self.HASHFN, strict=True
+        )
 
     def test_put(self):
         self.writer.put(b'dave', b'dave')
         self.assertEqual(self.get_reader().get(b'dave'), b'dave')
 
-        # Don't care about rich error, just as long as it crashes.
```

Comment:
These tests had a problem in that the reader was being called, which finalized the writer, but then we try to use the writer again. We were expecting exceptions, but not _that_ exception.